### ISSUE TITLE
Custom SVG Icon Support

### ISSUE THREAD

**Issue by ryancrosser** [primefaces/primevue#820]
Let me first start off by saying I am a big fan of Prime, first learning of and using PrimeNG back with AngularJS and more recently rediscovering PrimeVue! Thank you for your work.

Several years ago, the client I support pushed new group policies that blocked font icons forcing our developers to use SVG icons and that has become our standard ever since. I know the PrimeIcons are available in SVG, but I am not sure how I would use those SVG (or other SVGs) within the PrimeVue components, over the font icons. Currently, I am using [svgicon](https://mmf-fe.github.io/svgicon/en/) which allows me to use an SVG inside an `icon` tag, although that tag can be configured.

I was wondering if an additional prop could be added to the PrimeVue components, specifically menus and buttons, although it would be beneficial to have anywhere PrimeIcons are used, to accept either html (raw SVG) or, and more preferred, a component that would render the SVG icon, matching our current process?

In [issue 446](https://github.com/primefaces/primevue/issues/446), you mentioned overwriting the PrimeIcon CSS for FontAwesome, would that be possible with SVGs too?

**Comment by Dav3rs** [comment]
Yes, I want to know too. What is the recommended way to integrate an svg icons library like @mdi/js?

**Comment by ryancrosser** [comment]
While I would like to see better support provided directly within PrimeVue, I have found a workaround to use custom SVG icons. But I do not think this is not a great solution if you have a lot of icons that you need to use.

You can create a CSS class defining the `background` property referencing the .svg file with the `url` function. You can specify any other needed class properties in the same rule.

```
.custom-svg-icon {
  background: url("/path/to/svg/icon.svg");
  fill: red;
  height: 20px;
  width: 20px;
}

```

This class can now be used at a replacement for the `pi-` style icon classes provided by Prime Icons.

```
<Button icon="pi custom-svg-icon" class="p-button-text" />
```

This is working for me as I only needed a couple icons right now, but if, and when, I need additional icons, I do not see this as a long-term solution.

For material design icons, you may be able to use [@mdi/svg](https://www.npmjs.com/package/@mdi/svg) and directly link to the SVGs within the package, but using this technique, you would have to create a custom rule for each icon that you wish to use.

**Comment by cagataycivici** [comment]
Templating is the solution we apply to use SVG icons where available, an example from Button;

https://primefaces.org/primevue/showcase/#/button

Other components like menus will also receive templating support to use svgs and basically any other content.

**Comment by jayliu50** [comment]
@cagataycivici The templating idea as you suggested is interesting, but I hope is not the long-term solution.

But as a paying customer, I don't like the idea of having to completely override the template just to change an. Icons show up in so many places (columns of datatable, checkbox, button, etc), and it just seems heavy-handed to have to rewrite so many templates just to use a custom icon set. I do hope there is another way on the roadmap where we can hot-swap icons.

**Comment by FossPrime** [comment]
The primevue documentation for using your own svg icon needs improvement. The link @cagataycivici provided is broken. The [docs page](https://www.primefaces.org/primevue/button) has not a single demo or comment about using custom icons. 

For reference, Vuetify 3 does

```vue
<template>
<v-icon>{{ mdiAccountPlus }}</v-icon>
</template>
<script setup lang="ts">
  import { mdiAccountPlus } from '@mdi/js'
</script>
```

In global config you can change the default icons for components that use icons.

**Comment by 1gn0r4nd** [comment]
Issue seems to be closed. Anyone have an idea on what the current best practice is to add custom svg files? Do we fork https://github.com/primefaces/primeicons and add svg icons to it?

**Comment by cagataycivici** [comment]
The issue is closed because it applies to a wider scope. This year's focus is on customization so idea is "to make them yours not ours" by making them as flexible as possible so we'll provide templates for everywhere including icons so you can use anything inside not just svgs. This will be done while we're doing unstyled mode as they kinda overlap.

More info at;

https://www.primefaces.org/roadmap

**Comment by satezmedia** [comment]
We solved this by creating our own icon font from svgs with https://icomoon.io

**Comment by aDogCalledSpot** [comment]
It's not documented well but there is a template for the icon available. You can do the following:

```vue
<Button label="Button text">
  <template v-slot:icon>
    <svg ...></svg>
  </template>
</Button>
```

**Comment by beqachx** [comment]
I like how Vuetify solves this problem.

It would be perfect to have the ability to set **icon aliases**


Vuetify example:

```ts
// src/plugins/vuetify.ts
import { mdiAccount } from '@mdi/js'

export default createVuetify({
  icons: {
    aliases: {
      account: mdiAccount,
    },
  },
})
```

and then

```vue
<v-icon icon="$account" />
```

in the future, if you have to change the icon set, you will replace them only in the alias config


for more info: [view vuetify's custom icons and aliases](https://vuetifyjs.com/en/features/icon-fonts/#creating-a-custom-icon-set)
